Architecture/TodoApp/android/core/src/main/java/com/novoda/todoapp/tasks/service/PersistedTasksService.java

```diff
@@ -301,6 +306,104 @@ public void call() {
         };
     }
 
+    @Override
+    public void delete(final Task task) {
+
+        final long deleteActionTimestamp = clock.timeInMillis();
+
+        remoteDataSource.deleteTask(task.id())
+                .compose(deleteLocallyThenConfirmOrMarkAsError(task))
+                .flatMap(persistIfDeleteActionIsMostRecent(task, deleteActionTimestamp))
+                .subscribe(taskRelay);
+    }
+
+    private Observable.Transformer<Void, Event<Tasks>> deleteLocallyThenConfirmOrMarkAsError(final Task task) {
+        return asOrchestratedAction(new DataOrchestrator<Void, Event<Tasks>>() {
+            @Override
+            public Event<Tasks> startWith() {
+                Event<Tasks> currentState = taskRelay.getValue();
+                return currentState.asLoadingWithData(markTaskAsDeletedLocallyAt(currentState, task, clock.timeInMillis()));
+            }
+
+            @Override
+            public Event<Tasks> onConfirmed(Void value) {
+                return onError();
```

Comment:
We don't expect a value back there is two way you can get a confirmation, termination of the stream or termination of the stream with a value. In this case we never expect a value so it would be an error. Some other usages of this will be reversed where they need a value and a confirmation without one would be an error.